indexer_test.go

```diff
@@ -0,0 +1,82 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package main
+
+import (
+	"testing"
+
+	"github.com/elastic/package-registry/packages"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestLatestPackagesVersion(t *testing.T) {
+	newPackage := func(name string, version string) *packages.Package {
+		p := new(packages.Package)
+		p.Name = name
+		p.Version = version
+		return p
+	}
+
+	cases := []struct {
+		title    string
+		source   packages.Packages
+		expected packages.Packages
+	}{
+		{
+			title: "single package",
+			source: packages.Packages{
+				newPackage("foo", "1.2.3"),
+			},
+			expected: packages.Packages{
+				newPackage("foo", "1.2.3"),
+			},
+		},
+		{
+			title: "single package sorted versions",
+			source: packages.Packages{
+				newPackage("foo", "1.2.3"),
+				newPackage("foo", "1.2.2"),
+				newPackage("foo", "1.2.1"),
+			},
+			expected: packages.Packages{
+				newPackage("foo", "1.2.3"),
+			},
+		},
+		{
+			title: "single package unsorted versions",
```

Comment:
How about adding a case where both indexers contain the same package version?

Reply:
For the case where both indexers contain the same package version, what would the behavior be for a query with `?all=true`? 

I would think even in this case the response should only contain one item for that package version. The reason being that the other APIs don't provide a way to differentiate between the two items (e.g. the only specifiers are name and version in the download request `/epr/example/example-1.0.1.zip`). Assuming the solution is to only return one in the `search/?all=true` response then it would be good to specify somewhere what the priority order is.

Reply:
I guess that in that case we have to decide that some indexers have precendence over the others, I will take a look to this.

Reply:
I have added a change so indexes defined later have precedence over the ones defined earlier. I have made that because currently we are defining the zip indexer, with support for signatures, after the plain file indexer, so now if a package is in both indexers, the zip with signature will have precedence.